src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -369,21 +375,44 @@ export class EditorPart extends Part implements IEditorPart {
 				this.gridWidget.distributeViewSizes();
 				break;
 			case GroupsArrangement.MAXIMIZE:
-				this.gridWidget.maximizeViewSize(target);
-				break;
-			case GroupsArrangement.TOGGLE:
-				if (this.isGroupMaximized(target)) {
-					this.arrangeGroups(GroupsArrangement.EVEN);
-				} else {
-					this.arrangeGroups(GroupsArrangement.MAXIMIZE);
+				if (this.groups.length < 2) {
+					return; // need at least 2 groups to be maximized
 				}
-
+				this.disposableMaximizedView = this.gridWidget.maximizeView(target);
```

Comment:
I find it a bit strange to use a `IDisposable` for this purpose because disposables are for tracking to avoid memory leaks imho. If we stick with it then I would rename it to `unmaximizeViwHandler` or similar

Reply:
This concept is already used in for example the gridView. 

Also check this: https://github.com/microsoft/vscode/pull/194894#discussion_r1366958495

Reply:
Ok then maybe just rename it to indicate what it does.